File: assoc_arrays.py
# Initialize
translation = {"hello":"merhaba","goodbye":"hoscakal"}
print("\nThe dictionary after initialization: ")
print(translation)

# Get the value given for a given key
print("\nGetting the value of 'hello': " + translation["hello"])

# Add a new element
translation["one"] = 1
print("\nThe dictionary after 'one' is added: ")
print(translation)

# Remove an element
del(translation["one"])
print("\nThe dictionary after 'one' is deleted: ")
print(translation)

# Modify the value of an existing element
translation["hello"] = "selam"
print("\nThe dictionary after value of 'hello' is changed to 'selam': ")
print(translation)

# Search for the existence of a key
key_hello_exists = "hello" in translation
print("\nThe key 'hello' exists: " + str(key_hello_exists))

# Search for the existence of a value
value_selam_exists = "selam" in translation.values()
print("\nThe value 'selam' exists: " + str(value_selam_exists))

# Loop through an associative array
print("\nPrinting by iterating through the dictionary: ")
def foo(a, b):
	print(str(a) + ": " + str(b))

for key in translation:
	foo(key, translation[key])

File: assoc_arrays.rb
# Initialize
translate = {'hello' => 'merhaba', 'goodbye' => 'hoscakal'}
puts "The hash after initialization:"
puts translate

# Get the value for a given key
puts "The value for key 'hello':"
puts translate['hello']

# Add a new element
translate['one'] = 1
puts "The hash after 'one' is added:\n"
puts translate

# Remove an element
translate.delete('one')
puts "The hash after 'one' is deleted:"
puts translate

# Modify the value of an existing element
translate['hello'] = 'selam'
puts "The hash 'hello' is changed to 'selam':"
puts translate

# Search for the existence of a key
puts "\nThe key 'hello' exists:"
puts translate.key?("hello")

# Search for the existence of a value
 puts "The value 'selam' exists:"
 puts translate.value?("selam")

# Loop through hash
def foo(k, v)
    puts k + ": " + v + "\n"
end
puts "\n-The hash printed by iteration:\n"
translate.each do |key, array| 
    foo(key, translate[key])
end

